chromadb/utils/embedding_functions/llama_cpp_embedding_function.py

```diff
@@ -0,0 +1,91 @@
+from chromadb.api.types import Documents, EmbeddingFunction, Embeddings
+from typing import Any, cast
+
+class LlamaCppEmbeddingFunction(EmbeddingFunction):
+    
+    def __init__(self, model_path: str = "", huggingface_repo_id: str = "", huggingface_filename: str = "", **kwargs: Any) -> None:
```

Comment:
It makes sense, but since the `huggingface_filename` needs to use a * wildcard to search for the model correctly, I think it will be even easier to have the model path determine either the full path of the model or the hugging face repo id. This way the code checks will be simpler too.

```
To use a huggingface model: LlamaCppEmbeddingFunction(model_path="username/repo_name", huggingface_filename="*q8_0.gguf")
To use a local model: LlamaCppEmbeddingFunction(model_path="path/to/model")
```